solitaire.rb

```diff
@@ -1,2 +1,154 @@
+require_relative 'keystreamgenerator'
+
 class Solitaire
+
+  def initialize(opts={})
+    @message           = opts[:message]
+    @encrypted_message = opts[:encrypted_message]
+  end
+
+  def encrypt
+    sanitize_message
+    convert_message_to_numbers
+    convert_keystream_to_numbers
+    add_message_numbers
+    convert_to_letters(added_message_numbers)
+  end
+
+  def decrypt
+    convert_encrypted_message_to_numbers
+    convert_keystream_to_numbers
+    subtract_message_numbers
+    convert_to_letters(subtracted_message_numbers)
+  end
+
+  private
+
+  LETTER_CONVERTER = Hash[('A'..'Z').zip (1..26)]
+  NUMBER_CONVERTER = Hash[(1..26).zip ('A'..'Z')]
```

Comment:
I believe you can use Ruby's [Hash Invert][1] method here:

```ruby
h = { "n" => 100, "m" => 100, "y" => 300, "d" => 200, "a" => 0 }
h.invert   #=> {0=>"a", 100=>"m", 200=>"d", 300=>"y"}
```

[1]: http://ruby-doc.org/core-2.3.0/Hash.html#method-i-invert